addons/web/static/tests/views/fields/numeric_fields_tests.js

```diff
@@ -306,4 +310,64 @@ QUnit.module("Fields", (hooks) => {
             });
         }
     );
+
+    QUnit.test(
+        "useNumpadDecimal should synchronize handlers on input elements",
+        async function (assert) {
+            target = getFixture();
+            patchWithCleanup(localization, { decimalPoint: ",", thousandsSep: "." });
+
+            /**
+             * Takes an array of input elements and asserts that each has the correct event listener.
+             * @param {HTMLInputElement[]} inputEls
+             */
+            function testInputElements(inputEls) {
+                for (const inputEl of inputEls) {
+                    inputEl.focus();
+                    const numpadDecimalEvent = new KeyboardEvent("keydown", {
+                        code: "NumpadDecimal",
+                        key: ".",
+                    });
+                    numpadDecimalEvent.preventDefault = () => assert.step("preventDefault");
+                    inputEl.dispatchEvent(numpadDecimalEvent);
```

Comment:
though it is not really needed here, I think that - almost - every time we dispatch an event we should await the next tick.